src/main/java/org/elasticsearch/cluster/block/ClusterBlockLevel.java

```diff
@@ -44,13 +53,18 @@ public int id() {
         return this.id;
     }
 
-    public static ClusterBlockLevel fromId(int id) {
+    public static EnumSet<ClusterBlockLevel> fromId(int id) {
         if (id == 0) {
-            return READ;
+            return EnumSet.of(READ);
         } else if (id == 1) {
-            return WRITE;
+            return EnumSet.of(WRITE);
         } else if (id == 2) {
-            return METADATA;
+            // After 1.5.0 METADATA has been splitted into two distincts cluster block levels
+            return EnumSet.of(METADATA_READ, METADATA_WRITE);
+        } else if (id == 3) {
+            return EnumSet.of(METADATA_READ);
+        } else if (id == 4) {
+            return EnumSet.of(METADATA_WRITE);
```

Comment:
You're right. I added some logic in the ClusterBlock.writeTo() method, converting 3/4 values into 2 if the target node has a version <= 1.5.0.